src/laws/laws_utils.jl

```diff
@@ -0,0 +1,5 @@
+import Sleipnir: prepare_vjp_law
+
+prepare_vjp_law(simulation, law::ConstantLaw, law_cache, θ, glacier_idx) = nothing
+prepare_vjp_law(simulation, law::NullLaw, law_cache, θ, glacier_idx) = nothing
+prepare_vjp_law(simulation::Prediction, law::Law, law_cache, θ, glacier_idx) = nothing
```

Comment:
Is the idea that we will have to define this function for each law (inversion with respect to H->A, (H, T)->A)? As long as we want to use the cache, of course.